crates/hyperqueue/src/server/job.rs

```diff
@@ -185,16 +187,65 @@ impl Job {
         }
     }
 
-    pub fn make_job_detail(&self, include_tasks: bool) -> JobDetail {
-        let tasks = if include_tasks {
-            self.tasks.values().cloned().collect()
-        } else {
-            Vec::new()
+    pub fn make_job_detail(
+        &self,
+        task_id_selector: IdSelector,
+        task_status_selector: StatusSelector,
+    ) -> JobDetail {
+        let all_task_ids: Set<JobTaskId> = self.tasks.iter().map(|task| task.1.task_id).collect();
+        let mut task_ids_not_found: Vec<JobTaskId> = vec![];
+
+        // Select tasks by status
+        let mut tasks = match task_status_selector {
```

Comment:
To avoid going through the tasks multiple times, I would suggest to just iterate through them once and then check if they are valid (by status and ID). If yes, add them to a result vec (or add their ID to the not found vec).
That way we will also avoid needless clones if e.g. something passes through status but then is filtered by ID.